Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -1137,15 +1143,28 @@ private final class SQLiteManifestCache: Closable {
     }
 
     func put(key: ManifestLoader.ManifestCacheKey, manifest: ManifestLoader.ManifestParseResult) throws {
-        let query = "INSERT OR IGNORE INTO MANIFEST_CACHE VALUES (?, ?);"
-        try self.executeStatement(query) { statement -> Void in
-            let data = try self.jsonEncoder.encode(manifest)
-            let bindings: [SQLite.SQLiteValue] = [
-                .string(key.sha256Checksum),
-                .blob(data),
-            ]
-            try statement.bind(bindings)
-            try statement.step()
+        do {
+            let query = "INSERT OR IGNORE INTO MANIFEST_CACHE VALUES (?, ?);"
+            try self.executeStatement(query) { statement -> Void in
+                let data = try self.jsonEncoder.encode(manifest)
+                let bindings: [SQLite.SQLiteValue] = [
+                    .string(key.sha256Checksum),
+                    .blob(data),
+                ]
+                try statement.bind(bindings)
+                try statement.step()
+            }
+        } catch (let error as SQLite.Errors) where error == .databaseFull {
+            if !self.configuration.truncateWhenFull {
+                throw error
+            }
+            self.diagnosticsEngine?.emit(.warning("truncating manifest cache database since it reached max size of \(self.configuration.maxSizeInBytes ?? 0) bytes"))
+            try self.executeStatement("DELETE FROM MANIFEST_CACHE;") { statement -> Void in
```

Comment:
This looks as if it completely clears out the cache rather than truncating it using some LRU scheme; is that right?  Makes sense, just wanting to make sure I understand the behavior.  Thanks.

Reply:
yes there is no performant way to do LRU, so this is the tradeoff I made here

Reply:
That tradeoff makes complete sense, just make sure I didn't misunderstand (I think the work "truncate" initially made me misunderstand the behavior).